apollo-router/tests/integration/telemetry/metrics.rs

```diff
@@ -40,33 +40,55 @@ async fn test_metrics_reloading() {
         router.assert_reloaded().await;
     }
 
-    router.assert_metrics_contains(r#"apollo_router_cache_hit_count_total{kind="query planner",storage="memory",otel_scope_name="apollo/router"} 4"#, None).await;
-    router.assert_metrics_contains(r#"apollo_router_cache_miss_count_total{kind="query planner",storage="memory",otel_scope_name="apollo/router"} 2"#, None).await;
-    router.assert_metrics_contains(r#"apollo_router_http_request_duration_seconds_bucket{status="200",otel_scope_name="apollo/router",le="100"}"#, None).await;
-    router
-        .assert_metrics_contains(r#"apollo_router_cache_hit_time"#, None)
-        .await;
-    router
-        .assert_metrics_contains(r#"apollo_router_cache_miss_time"#, None)
-        .await;
-    router
-        .assert_metrics_contains(r#"apollo_router_session_count_total"#, None)
-        .await;
-    router
-        .assert_metrics_contains(r#"custom_header="test_custom""#, None)
-        .await;
+    let metrics = router
+        .get_metrics_response()
+        .await
+        .expect("failed to fetch metrics")
+        .text()
+        .await
+        .unwrap();
+
+    check_metrics_contains(
+        &metrics,
+        r#"apollo_router_cache_hit_count_total{kind="query planner",storage="memory",otel_scope_name="apollo/router"} 4"#,
+    );
+    check_metrics_contains(
+        &metrics,
+        r#"apollo_router_cache_miss_count_total{kind="query planner",storage="memory",otel_scope_name="apollo/router"} 2"#,
+    );
+    check_metrics_contains(
+        &metrics,
+        r#"apollo_router_http_request_duration_seconds_bucket{status="200",otel_scope_name="apollo/router",le="100"}"#,
+    );
+    check_metrics_contains(&metrics, r#"apollo_router_cache_hit_time"#);
+    check_metrics_contains(&metrics, r#"apollo_router_cache_miss_time"#);
+    check_metrics_contains(&metrics, r#"apollo_router_session_count_total"#);
+    check_metrics_contains(&metrics, r#"custom_header="test_custom""#);
+    check_metrics_contains(&metrics, r#"apollo_router_cache_hit_time"#);
```

Comment:
Seems to be repeating the test for `apollo_router_cache_hit_time` which is at line 63.

```suggestion
```